### DISCUSSION TITLE
Are you a vendor who needs help calling the Selling Partner API using Python?

### DISCUSSION THREAD

**Discussion by Guerri114** [amzn/selling-partner-api-models#3116] [General]
If you are a vendor who wants to use the Selling Partner API, the attached Python scripts should help you get started once you register for access in Vendor Central and have the necessary credentials.

The attached code samples are examples of how to exchange your refresh token for an access token, sign your request, and call the API. There are two files in the .zip: one file is for if you set up your Selling Partner API application with an AWS IAM Role and need to sign the request by assuming that role. The other file is for if you set up your Selling Partner API application with an AWS IAM User, which doesn't involve the added step of assuming a role.

Both files call the [getCatalogItem API operation](https://github.com/amzn/selling-partner-api-docs/blob/main/references/catalog-items-api/catalogItems_2020-12-01.md#getcatalogitem), one of many APIs and API operations in the Selling Partner API. I struggled for a while to get things to work in Python, so I hope this helps someone else out.
[SPAPI-getCatalogItem.zip](https://github.com/amzn/selling-partner-api-docs/files/6994447/SPAPI-getCatalogItem.zip)

**Comment by ash-singh-31** [comment]
Hi @Guerri114 , Are you able to get the data for Sales Diagnostic Report via API which we use in Vendor Central ? If yes, I need some of your help. Please respond.

**Reply by Guerri114** [reply]
@ash-singh-31 : Yes, you can get the Sales Diagnostic report via API as well as many other reports. You should review the following documentation:

- [Reports API Use Case Guide](https://developer-docs.amazon.com/sp-api/docs/reports-api-v2021-06-30-use-case-guide)
- [Report Type values](https://developer-docs.amazon.com/sp-api/docs/report-type-values) - you'll want to focus on **vendor retail analytics** and **brand analytics** and **promotion performance report**

If you've never used the Selling Partner API before, be sure to read the [Selling Partner API Developer Guide](https://developer-docs.amazon.com/sp-api/docs/what-is-the-selling-partner-api)

**Reply by ash-singh-31** [reply]
Hi @Guerri114 Thank you for your quick reply. Correct me if I am wrong. Since we are Vendor to the Amazon, do we need to registered in Amazon's Brand Registry ?

**Reply by Guerri114** [reply]
In order to access the reports via API, yes, as the documentation says, you must be registered in Amazon's [Brand Registry](https://brandservices.amazon.com/).
